extensions/mode-control-register.md

```diff
@@ -0,0 +1,53 @@
+# Overview
+
+This document describes the mode control register in a centralized location so that other extensions which use it can refer to a single document
+instead of duplicating it between all relevant extensions. The mode control register adds a new concept of _modes_ to the ISA. Modes are used to
+control things about the processor state which are binary-incompatible - that is, code written for a mode other than the mode the
+processor is actually in will almost certainly behave incorrectly.
+
+# Modes
+
+CR `1 0001`, equivalently `cr17` or "the MODE control register," holds a value indicating the current
+_mode_ of the processor. This has an initial value of `0`.
+
+## Base Mode (Real 16-bit Address Mode)
+
+The mode described by [the base isa](../../base-isa.md) is known as the
+Base mode. In that mode, pointers are 16 bits and there is no virtual memory, paging, or memory protection.
+Base mode is indicated by a value of 0 in `cr17`.
+
+## Real n-bit Address Mode
+
+In this mode, all addresses are treated as being `n` bits long. Addresses always refer to their sign extension to the highest supported address size.
+Entering this mode from a lower n-bit address mode _must_ preserve the program counter. Returning to a lower n-bit address mode must preserve the program
+counter _if possible_ - if the current program counter interpreted as the lower n-bit Address would not refer to the same location, then the behavior of
+the system is _unspecified_.
+
+# Address Extension Table
+
+| Mode CR Value | Address Size | Extension                                                   |
+|---------------|--------------|-------------------------------------------------------------|
+| 0             | 16 bit       | [Base ISA](../base-isa.md)                                  |
+| 2             | 32 bit       | [Real 32-bit Address Space](32-bit-address-space/README.md) |
+| 4             | 64 bit       | [Real 64-bit Address Space](64-bit-address-space/README.md) |
+
+# Interactions With Other Extensions
+
+Other extensions mention "address modes", in particular when deciding what a particular value
+should mean or how many bytes an immediate should be.
+
+| Address Size | Address Mode |
+|--------------|--------------|
+| 16 bit       | word         |
+| 32 bit       | doubleword   |
+| 64 bit       | quadword     |
+
+If the system supports [privilege levels](../privileged-mode/), then `cr17` is only writable when in system privilege mode.
+
+# Recommendations
```

Comment:
I would guess this section still best belongs into the individual extensions? Since later extensions make it potentially obsolete. Otherwise the wording should be changed slightly.

Reply:
Why would the privilege-ness of the control register be better in each extension?

Reply:
Sorry, wrong segment selection. I mean the recommendations section.